app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -117,12 +117,18 @@ def start_clone(clone_options)
 
     [:config, :customization, :linked_clone].each { |key| vim_clone_options[key] = clone_options[key] }
 
-    [:folder, :host, :pool, :snapshot].each do |key|
+    [:folder, :host, :pool].each do |key|
       ci = clone_options[key]
       next if ci.nil?
+
       vim_clone_options[key] = ci.ems_ref_obj
     end
 
+    if clone_options[:snapshot]
+      ci = clone_options[:snapshot]
+      vim_clone_options[:snapshot] = VimString.new(ci.ems_ref, ci.ems_ref_type, :ManagedObjectReference) if ci.ems_ref.present? && ci.ems_ref_type.present?
+    end
```

Comment:
NOTE per description this is temporary until we can STI Snapshots and include the EmsRefObjMixin normally